api/payments.js

```diff
@@ -113,89 +80,53 @@ function formatPaymentHelper(account, transaction, callback) {
  */
 function submitPayment(account, payment, clientResourceID, secret,
     urlBase, options, callback) {
-  var self = this;
-  var max_fee = Number(options.max_fee) > 0 ?
-    xrpToDrops(options.max_fee) : undefined;
-  var fixed_fee = Number(options.fixed_fee) > 0 ?
-    xrpToDrops(options.fixed_fee) : undefined;
-
-  // these checks are done at the top of the function because we cannot
-  // raise exceptions inside callbacks
-  validate.addressAndMaybeSecret({address: account, secret: secret});
-  validate.payment(payment);
-  validate.client_resource_id(clientResourceID);
-  validate.options(options);
-
-  function formatTransactionResponse(message, meta, _callback) {
-    if (meta.state === 'validated') {
-      var transaction = message.tx_json;
-      transaction.meta = message.metadata;
-      transaction.validated = message.validated;
-      transaction.ledger_index = transaction.inLedger = message.ledger_index;
 
-      return formatPaymentHelper(payment.source_account, transaction,
-                                 _callback);
+  function formatTransactionResponse(message, meta) {
+    if (meta.state === 'validated') {
+      var txJSON = message.tx_json;
+      txJSON.meta = message.metadata;
+      txJSON.validated = message.validated;
+      txJSON.ledger_index = txJSON.inLedger = message.ledger_index;
+      return formatPaymentHelper(payment.source_account, txJSON);
     }
 
-    _callback(null, {
+    return {
       client_resource_id: clientResourceID,
       status_url: urlBase + '/v1/accounts/' + payment.source_account
         + '/payments/' + clientResourceID
-    });
+    };
   }
 
-  function _createPaymentTransaction(remote, _callback) {
-    var transaction;
-    try {
-      transaction = createPaymentTransaction(account, payment);
-    } catch (err) {
-      _callback(err);
-      return;
-    }
-
-    var ledgerIndex;
-    var maxFee = Number(max_fee);
-    var fixedFee = Number(fixed_fee);
+  function prepareTransaction(_transaction, remote) {
+    validate.client_resource_id(clientResourceID);
 
-    if (Number(options.last_ledger_sequence) > 0) {
-      ledgerIndex = Number(options.last_ledger_sequence);
-    } else {
-      ledgerIndex = Number(remote._ledger_current_index)
-        + transactions.DEFAULT_LEDGER_BUFFER;
-    }
-
-    transaction.lastLedger(ledgerIndex);
+    _transaction.lastLedger(Number(options.last_ledger_sequence) ||
```

Comment:
Since `0`, `NaN` and negative works are falsy, this works. Consider casting to Boolean for clarity